tools/testkit/src/fevm.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use fvm::executor::{ApplyKind, ApplyRet, Executor};
+use fvm_ipld_encoding::tuple::*;
+use fvm_ipld_encoding::{strict_bytes, BytesSer, RawBytes};
+use fvm_shared::address::Address;
+use fvm_shared::message::Message;
+use fvm_shared::{ActorID, METHOD_CONSTRUCTOR};
+
+use crate::{Account, BasicTester};
+
+pub fn create_contract(tester: &mut BasicTester, owner: &mut Account, contract: &[u8]) -> ApplyRet {
+    let create_msg = Message {
+        from: owner.account.1,
+        to: Address::new_id(10), // EAM
+        gas_limit: 10_000_000_000,
+        method_num: EAMMethod::CreateExternal as u64,
+        params: RawBytes::serialize(BytesSer(contract)).unwrap(),
+        sequence: owner.seqno,
+        ..Message::default()
+    };
+    let create_mlen = create_msg.params.len();
+
+    let create_res = tester
+        .executor
+        .as_mut()
+        .unwrap()
+        .execute_message(create_msg, ApplyKind::Explicit, create_mlen)
+        .unwrap();
+
+    owner.seqno += 1;
+    create_res
+}
+
+pub fn invoke_contract(
+    tester: &mut BasicTester,
+    src: &mut Account,
+    dest: Address,
+    input_data: &[u8],
+    gas: i64,
+) -> ApplyRet {
+    let invoke_msg = Message {
+        from: src.account.1,
+        to: dest,
+        sequence: src.seqno,
+        gas_limit: gas,
+        method_num: EVMMethod::InvokeContract as u64,
+        params: RawBytes::serialize(BytesSer(input_data)).unwrap(),
+        ..Message::default()
+    };
+    let invoke_mlen = invoke_msg.params.len();
+
+    let invoke_res = tester
+        .executor
+        .as_mut()
+        .unwrap()
+        .execute_message(invoke_msg, ApplyKind::Explicit, invoke_mlen)
+        .unwrap();
+
+    src.seqno += 1;
```

Comment:
Does this always need to be incremented? I use this method to query the balance in SimpleCoin, which is a `view`, so it can be run locally without sending a transaction. Shouldn't the nonce stay the same?